lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -788,6 +797,11 @@ def from_ansible_params(ap, old_service, image_digest):
         if ap['force_update']:
             s.force_update = int(str(time.time()).replace('.', ''))
 
+        if ap['groups'] is not None:
+            # In case integers are passed as groups, we need to convert them to
+            # strings as docker internally treats them as strings.
+            s.groups = [str(g) for g in ap['groups']]
```

Comment:
Hopefully (once #50335 is merged) this won't be necessary anymore, since Ansible will convert the list elements to strings automatically. But for now, this is still necessary.

Reply:
Yeah that one caught me a bit of guard as I was sure that it was the default behavior.